contracts/referrals/ReferralGateway.sol

```diff
@@ -719,6 +659,135 @@ contract ReferralGateway is
     }
 
+    function _payContribution(
+        uint256 _contribution,
+        string calldata _tDAOName,
+        address _parent,
+        address _newMember,
+        uint256 _couponAmount
+    ) internal whenNotPaused nonReentrant returns (uint256 _finalFee, uint256 _discount) {
+        uint256 realContribution;
+
+        if (_couponAmount > _contribution) realContribution = _couponAmount;
+        else realContribution = _contribution;
+
+        _payContributionChecks(realContribution, _tDAOName, _parent, _newMember);
+
+        _finalFee = (realContribution * SERVICE_FEE_RATIO) / 100;
+        // If the DAO pre join is enabled it means the DAO is not deployed yet
+        if (nameToDAOData[_tDAOName].preJoinEnabled) {
+            // The prepaid member object is created inside this if statement only
+
+            // It will get a discount as a pre-joiner
+            _discount +=
+                ((realContribution - _couponAmount) * CONTRIBUTION_PREJOIN_DISCOUNT_RATIO) /
+                100;
+            uint256 toReferralReserve;
+
+            if (nameToDAOData[_tDAOName].referralDiscount) {
+                toReferralReserve = (realContribution * REFERRAL_RESERVE) / 100;
+
+                if (_parent != address(0)) {
+                    uint256 referralDiscount = ((realContribution - _couponAmount) *
+                        REFERRAL_DISCOUNT_RATIO) / 100;
+                    _discount += referralDiscount;
+
+                    childToParent[_newMember] = _parent;
+
+                    (_finalFee, nameToDAOData[_tDAOName].referralReserve) = _parentRewards(
+                        _newMember,
+                        realContribution,
+                        nameToDAOData[_tDAOName].referralReserve,
+                        toReferralReserve,
+                        _finalFee,
+                        _tDAOName
+                    );
+                } else {
+                    nameToDAOData[_tDAOName].referralReserve += toReferralReserve;
+                }
+            }
+
+            uint256 rePoolFee = (realContribution * REPOOL_FEE_RATIO) / 100;
+
+            _finalFee -= _discount + toReferralReserve + rePoolFee;
+
+            assert(
+                (realContribution * SERVICE_FEE_RATIO) / 100 ==
+                    _finalFee + _discount + toReferralReserve + rePoolFee
+            );
+
+            nameToDAOData[_tDAOName].toRepool += rePoolFee;
+            nameToDAOData[_tDAOName].currentAmount +=
+                realContribution -
+                (realContribution * SERVICE_FEE_RATIO) /
+                100;
+            nameToDAOData[_tDAOName].collectedFees += _finalFee;
+
+            uint256 amountToTransfer = realContribution - _discount - _couponAmount;
+
+            if (amountToTransfer > 0) {
+                usdc.safeTransferFrom(_newMember, address(this), amountToTransfer);
+                usdc.safeTransfer(operator, _finalFee);
```

Comment:
Transferring to the operator should be after the transfer from the coupon pool. The Op fees are always there even if the coupon covers everything.

Reply:
Commit 961c861 and tests in commit f5f7b0b